exporter/opentelemetry-exporter-otlp-proto-common/src/opentelemetry/exporter/otlp/proto/common/metrics_encoder/__init__.py

```diff
@@ -0,0 +1,237 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+from typing import Sequence, Any, Mapping, Dict, Callable, List
+
+from opentelemetry.proto.collector.metrics.v1.metrics_service_pb2 import (
+    ExportMetricsServiceRequest,
+)
+from opentelemetry.proto.common.v1.common_pb2 import (
+    KeyValue,
+    InstrumentationScope,
+    AnyValue,
+    ArrayValue,
+    KeyValueList,
+)
+from opentelemetry.proto.metrics.v1 import metrics_pb2 as pb2
+from opentelemetry.sdk.metrics._internal.point import (
+    MetricsData,
+    Gauge,
+    Histogram,
+    Sum,
+)
+from opentelemetry.proto.resource.v1.resource_pb2 import (
+    Resource as PB2Resource,
+)
+from opentelemetry.sdk.resources import Resource
+
+_logger = logging.getLogger(__name__)
+
+
+def _translate_attributes(attributes) -> Sequence[KeyValue]:
+    output = []
+    if attributes:
+
+        for key, value in attributes.items():
+            try:
+                output.append(_translate_key_values(key, value))
+            except Exception as error:  # pylint: disable=broad-except
+                _logger.exception(error)
+    return output
+
+
+def encode_metrics(data: MetricsData) -> ExportMetricsServiceRequest:
+    resource_metrics_dict = {}
+
+    for resource_metrics in data.resource_metrics:
+
+        resource = resource_metrics.resource
+
+        # It is safe to assume that each entry in data.resource_metrics is
+        # associated with an unique resource.
+        scope_metrics_dict = {}
+
+        resource_metrics_dict[resource] = scope_metrics_dict
+
+        for scope_metrics in resource_metrics.scope_metrics:
+
+            instrumentation_scope = scope_metrics.scope
+
+            # The SDK groups metrics in instrumentation scopes already so
+            # there is no need to check for existing instrumentation scopes
+            # here.
+            pb2_scope_metrics = pb2.ScopeMetrics(
+                scope=InstrumentationScope(
+                    name=instrumentation_scope.name,
+                    version=instrumentation_scope.version,
+                )
+            )
+
+            scope_metrics_dict[instrumentation_scope] = pb2_scope_metrics
+
+            for metric in scope_metrics.metrics:
+                pb2_metric = pb2.Metric(
+                    name=metric.name,
+                    description=metric.description,
+                    unit=metric.unit,
+                )
+
+                if isinstance(metric.data, Gauge):
+                    for data_point in metric.data.data_points:
+                        pt = pb2.NumberDataPoint(
```

Comment:
Include `start_time_unix_nano`?

Reply:
Did I remove this or is this an addition?

Reply:
Doesn't this exist already?

Reply:
It does for metrics Histogram and Sum, but was never given to the Guage at least in the grpc and http separate implementations

Reply:
If we agree on this change, I would just make a tracking issue as to not include any more additions to the PR :)

Reply:
I compared this side by side, we don't need to make any new additions here.